Runtime reload configuration

inserve-paul opened the pull request:
See discussion: https://github.com/spatie/laravel-backup/discussions/1900

Added an option to reload the configuration. For e.g. when used in a multi tenancy environment where the command is called from another command and the config has been changed for the specific tenant.

inserve-paul commented:
@freekmurze Would you like to take a look at this PR? Thank you!

steffensatrapa commented:
Thanks for your PR — it pointed me in the right direction. 
I have several backup configurations that I used to load dynamically in a BackupJob. However, after upgrading to v9, this no longer worked.

Here are a couple of suggestions:

1) Extend the --reload-config=<configname> option — this is what I’m using now to load specific configurations:

``` 
protected $signature = 'configurable-backup:run {--filename=} {--only-db} {--db-name=*} {--only-files} {--only-to-disk=} {--disable-notifications} {--timeout=} {--tries=} {--reload-config=}';
...
public function handle(): int
{
    if ($this->option('reload-config')) {
        $this->config = Config::fromArray(config($this->option('reload-config') ?? 'backup'));
    }
     ...
}
```
2) Don’t forget to extend the other backup commands as well, like backup:clean, which I also had to modify.

inserve-paul commented:
> Could you add a test to prove that the option works. Could you also update the docs?

@freekmurze I've updated the PR, added documentation and a test

freekmurze commented on docs/taking-backups/overview.md:
```diff
@@ -104,7 +104,24 @@ This section of the configuration determines which files and databases will be b
+
+foreach ($tenants as $tenant) {
+    config(['backup.backup.destination.filename_prefix' => $tenant.'_']);
+    $this->call('backup:run', ['--timeout' => 120, '--config' => 'backup']);
```
I'm afraid I don't understand the example fully. 

I was expecting that you could use another back config with something like this

php artisan backup:run --config=tenant1

steffensatrapa replied:
maybe better (my use-case):

### Running Backups with Specific Configuration

If you want to back up different areas of your Laravel application separately – for example with different schedules, database connections, filesystem disks, or cleanup settings – you can create custom backup configuration files.

#### Example:

Additional config files placed in the `config/` directory:

- `config/backup_database.php`
- `config/backup_invoices.php`
- `config/backup_uploads.php`

You can then run backups and cleanup commands individually:

```bash
php artisan backup:run --config=backup_database
php artisan backup:clean --config=backup_database

php artisan backup:run --config=backup_invoices
php artisan backup:clean --config=backup_invoices

php artisan backup:run --config=backup_uploads
php artisan backup:clean --config=backup_uploads
```
This allows full flexibility in scheduling, retention, and target destinations for each backup scope.

inserve-paul replied:
@steffensatrapa Thanks for your addition!

@freekmurze Is that example clear? In that case I will add it to the documentation

zfm-andreasmatuszewski replied:
Any updates here? @freekmurze

websanova replied:
Just came across this and need this as well, the `--config` option looks great to me, when is this getting merged?
